test/JDBC/expected/BABEL-2902.out

```diff
@@ -0,0 +1,548 @@
+
+SET babelfish_showplan_all off
+SELECT DB_NAME() --should be master
+GO
+~~START~~
+nvarchar
+babel_2902_testdb_1
+~~END~~
+
+
+SET babelfish_showplan_all on
+GO
+USE babel_2902_testdb_1
+PRINT 'test'
+PRINT 'test'
+USE master
+USE babel_2902_testdb_2
+PRINT 'test'
+PRINT 'test'
+SELECT FROM WHERE
+USE babel_2902_testdb_3 --should not be reached
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: syntax error near 'FROM' at line 8 and character position 7)~~
+
+SET babelfish_showplan_all off
+SELECT DB_NAME() --should be master
+GO
+~~START~~
+nvarchar
+babel_2902_testdb_1
+~~END~~
+
+
+
+---- TEST RAISEERROR ----
+-- note that during explain SQL Server does very limited no validation on inputs to raiserror 
+-- This means that normally invalid cases such as using error code that are too low, too high
+-- or are non existant do not result in an error. Similarly, things like argument counts are 
+-- not validated, nor are variables checked for existance, so long as it is syntactically correct
+-- explain will accept the raiserror statements
+SET babelfish_showplan_all ON
+GO
+--happy path--
+RAISERROR (N'TEST %s %d.',1,1,N'testing',9);
+GO
+~~START~~
+text
+Query Text: RAISERROR (N'TEST %s %d.',1,1,N'testing',9)
+~~END~~
+
+--system error message, note message code validation is NOT done during explain--
+RAISERROR (10000,1,1,N'testing',9);
+GO
+~~START~~
+text
+Query Text: RAISERROR (10000,1,1,N'testing',9)
+~~END~~
+
+--variables, note that variable substitution is NOT done during explain --
+DECLARE @testMsg VARCHAR(100)
+SET @testMsg = 'a: %s b: %s c :%s'
+RAISERROR (@testMsg, 1, 1, 'a', 'b', 'c')
+GO
+~~START~~
+text
+Query Text: ASSIGN @testmsg = SELECT 'a: %s b: %s c :%s'
+  Query Text: SELECT 'a: %s b: %s c :%s'
+  ->  Result  (cost=0.00..0.01 rows=1 width=32)
+Query Text: RAISERROR (@testMsg,1,1,'a','b','c')
+~~END~~
+
+
+-- TEST THROW --
+-- Throw is similar to RAISERROR in that during explain it is lax in enforcement of validation
+-- However it does validate two cases in particular. First if it is invoked without parameters
+-- you must be inside a catch block. arguments must be of the correct types
+--happy path--
+THROW 51000, 'test', 1
+GO
+~~START~~
+text
+Query Text: THROW 51000, 'test', 1
+~~END~~
+
+-- invalid argument types --
+THROW 'a', 'test', 1
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: procedure throw(unknown, unknown, integer) does not exist)~~
+
+THROW 51000, 1, 1
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: procedure throw(integer, integer, integer) does not exist)~~
+
+THROW 51000, 1, 'test'
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: procedure throw(integer, integer, unknown) does not exist)~~
+
+-- try catch --
+THROW
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: THROW without parameters should be executed inside a CATCH block)~~
+
+
+-- Unable to test until explain for RESTORE CTX PARTIAL is implemented
```

Comment:
Can we uncomment the below? It's worth to add a test case even throwing unsupported errors because it can guide other developers when they implement a new feature.